erasure-coding/src/lib.rs

```diff
@@ -124,8 +125,13 @@ fn code_params(n_validators: usize) -> Result<CodeParams, Error> {
 /// Obtain erasure-coded chunks, one for each validator.
 ///
 /// Works only up to 65536 validators, and `n_validators` must be non-zero.
-pub fn obtain_chunks(n_validators: usize, block_data: &BlockData, outgoing: &OutgoingMessages)
-	-> Result<Vec<Vec<u8>>, Error>
+pub fn obtain_chunks(
+	n_validators: usize,
+	relay_parent: H256,
+	candidate_hash: H256,
+	block_data: &BlockData,outgoing: &Option<AvailableMessages>
```

Comment:
formatting

Reply:
The `AvailableMessages` struct contains the queue root hashes, which is redundant with information that's already in the `Candidate`. We probably just want a separate wrapper around `Vec<Vec<u8>>` where the size of the outer vec == size of `candidate.egress_queue_roots`, and 

`message_queue_root(messages[i]) == egress_queue_roots[i].1`